tempesta_fw/str.c

```diff
@@ -668,6 +669,34 @@ tfw_str_next_str_val(const TfwStr *str)
 }
 EXPORT_SYMBOL(tfw_str_next_str_val);
 
+/**
+ * Function for crc32 calculation from TfwStr object.
+ * May be called from interrupt context.
+ */
+int
+tfw_str_crc32_calc(const TfwStr *str, u32 *crc32)
+{
+	const TfwStr *c, *end;
+	char *p, *data;
+	unsigned long len = 0;
+
+	BUG_ON(str->len && !str->ptr);
+	data = p = kmalloc(str->len, GFP_ATOMIC);
```

Comment:
Why not use `tfw_str_to_cstr()` here?

Reply:
Yes, missed that handy function. Thanks.
Corrected.